packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubble.base.tsx

```diff
@@ -62,14 +67,20 @@ export class TeachingBubbleBase extends BaseComponent<ITeachingBubbleProps, ITea
     };
 
     const classNames = getClassNames(styles, stylesProps);
+    const calloutStyles = classNames.subComponentStyles
+      ? ((classNames.subComponentStyles as ITeachingBubbleSubComponentStyles).callout as IStyleFunctionOrObject<
```

Comment:
Couldn't get around this :( cc: @JasonGore

Reply:
I'm honestly a bit confused because when I [tried to type `ITextFieldSubComponentStyles`](https://github.com/OfficeDev/office-ui-fabric-react/blob/3f75dd975a669b877023aaf45dd1c711c85f0e59/packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts#L324) I got a TS error, which also required me to do [some casting](https://github.com/OfficeDev/office-ui-fabric-react/blob/3f75dd975a669b877023aaf45dd1c711c85f0e59/packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx#L375) similar to what you're doing here.

But here I'm confused because:
1. If you're able to type `ITeachingBubbleSubComponentStyles`, I wonder why you have to do the first cast?
2. Even if you have to do the first cast, the type of `callout` should be properly inferred.

I'd be curious what the specific errors are when leaving out each cast, particularly the second one which seems especially redundant.

Reply:
> 
> 
> I'm honestly a bit confused because when I [tried to type `ITextFieldSubComponentStyles`](https://github.com/OfficeDev/office-ui-fabric-react/blob/3f75dd975a669b877023aaf45dd1c711c85f0e59/packages/office-ui-fabric-react/src/components/TextField/TextField.types.ts#L324) I got a TS error, which also required me to do [some casting](https://github.com/OfficeDev/office-ui-fabric-react/blob/3f75dd975a669b877023aaf45dd1c711c85f0e59/packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx#L375) similar to what you're doing here.
> 
> But here I'm confused because:
> 
>     1. If you're able to type `ITeachingBubbleSubComponentStyles`, I wonder why you have to do the first cast?

```shell
[ts] Property 'callout' does not exist on type '{}'. [2339]
```

>     2. Even if you have to do the first cast, the type of `callout` should be properly inferred.

Not an issue if only the first cast is done. Is that preferable?

```tsx
    const calloutStyles = classNames.subComponentStyles
      ? (classNames.subComponentStyles as ITeachingBubbleSubComponentStyles).callout
      : undefined;
```

> 
> I'd be curious what the specific errors are when leaving out each cast, particularly the second one which seems especially redundant.

Reply:
Thanks for looking at this :)

Reply:
Yeah, at the very least the second cast shouldn't need to be done since the first cast's `callout` type matches what you expect.

Reply:
If you switch to TS3.1 in VSCode, does your first error go away?

Reply:
> 
> 
> If you switch to TS3.1 in VSCode, does your first error go away?

Same error as before unfortunately. But I'm not blocked by it.

```shell
tsc -v
Version 3.1.6
```

Reply:
Hmm, actually I may be blocked on whatever is going on here judging from CI output...

Previous commit's [build output](https://travis-ci.org/OfficeDev/office-ui-fabric-react/builds/460491588?utm_source=github_status&utm_medium=notification):

```shell
../../packages/office-ui-fabric-react/lib/components/TeachingBubble/TeachingBubble.types.d.ts:26:64 - error TS2344: Type 'ITeachingBubbleStyles' does not satisfy the constraint 'IStyleSet<ITeachingBubbleStyles>'.
  Type 'ITeachingBubbleStyles' is not assignable to type '{ subComponentStyles?: { callout: IStyleFunctionOrObject<any, IStyleSet<any>>; }; }'.
    Types of property 'subComponentStyles' are incompatible.
      Type 'ITeachingBubbleSubComponentStyles' is not assignable to type '{ callout: IStyleFunctionOrObject<any, IStyleSet<any>>; }'.
        Types of property 'callout' are incompatible.
          Type 'IStyleFunctionOrObject<ICalloutContentStyleProps, ICalloutContentStyles>' is not assignable to type 'IStyleFunctionOrObject<any, IStyleSet<any>>'.
            Type 'IStyleFunction<ICalloutContentStyleProps, ICalloutContentStyles>' is not assignable to type 'IStyleFunctionOrObject<any, IStyleSet<any>>'.
              Type 'IStyleFunction<ICalloutContentStyleProps, ICalloutContentStyles>' has no properties in common with type 'Partial<IStyleSet<any>>'.
26     styles?: IStyleFunctionOrObject<ITeachingBubbleStyleProps, ITeachingBubbleStyles>;
```

Reply:
This may have been what required me to do the `<any, any>` approach I linked above, but I don't remember exactly.

Reply:
Sorry, I missed that link above. I had come across that when implemented `subComponentStyles` and thought I could get away without using it... _I was wrong_. Thank you for your help 👍.